### ISSUE TITLE
Error: malformed MAC address

### ISSUE THREAD

**Issue by mhuell** [AlexGustafsson/homebridge-wol#131]
### Backstory
I've tried to start my PC over the homebridge-wol v4.3.0 plugin. Error massage occours

### Issue
I get the error: malformed MAC address '‎<mac-address>'

### Environment
* os & os version: Raspbian GNU/Linux Buster (10) 
* node version: v14.15.1
* homebridge version: v1.1.6 
* homebridge-wol version: v4.3.0

### Configuration
My configuration looks like this:
"accessories": [
        {
            "name": "PC",
            "ip": "1.1.0.10",
            "pingInterval": 2,
            "pingsToChange": 5,
            "pingTimeout": 1,
            "pingCommandTimeout": 0,
            "mac": "<mac-address>",
            "broadcastAddress": "1.1.0.255",
            "startCommandTimeout": 0,
            "wakeGraceTime": 20,
            "wakeCommandTimeout": 0,
            "shutdownGraceTime": 15,
            "shutdownCommandTimeout": 0,
            "log": true,
            "logPinger": false,
            "debugLog": false,
            "returnEarly": false,
            "accessory": "NetworkDevice"
        }

Please paste your homebridge config.json here

IMPORTANT!
Remove any mac addresses from the configuration by
exchanging them with <mac-address>
```

### Log
When I follow these steps:
1. Make sure `logLevel` is set to `Debug` on all `NetworkDevice` accessories
2. Run `homebridge`
3. Let it run for a couple of minutes
4. Try to turn a device on or off using the plugin
5. Wait one minute
6. Repeat step 4

I get the following log:
```
Please paste your log here
[21/11/2020, 19:53:01] [PC] Waiting for wake grace time (20000ms) for "PC" (1.1.0.10)
[21/11/2020, 19:53:22] [PC] NetworkDevice "PC" (1.1.0.10) went from status "Waking Up" to "Offline"
[21/11/2020, 19:53:28] [PC] NetworkDevice awake cycle started for "PC" (1.1.0.10)
[21/11/2020, 19:53:28] [PC] Attempting to wake up "PC" (1.1.0.10)
[21/11/2020, 19:53:28] [PC] NetworkDevice "PC" (1.1.0.10) went from status "Offline" to "Waking Up"
[21/11/2020, 19:53:28] [PC] Attempting to start "PC" (1.1.0.10) by sending a WoL packet
[21/11/2020, 19:53:28] [PC] An error occured while trying to start "PC" (1.1.0.10): Error: malformed MAC address '‎<mac-address>'
    at Object.exports.createMagicPacket (/usr/local/lib/node_modules/homebridge-wol/node_modules/wake_on_lan/wake_on_lan.js:20:11)
    at Object.exports.wake (/usr/local/lib/node_modules/homebridge-wol/node_modules/wake_on_lan/wake_on_lan.js:50:30)
    at /usr/local/lib/node_modules/homebridge-wol/lib/util.js:5:9
    at new Promise (<anonymous>)
    at wake (/usr/local/lib/node_modules/homebridge-wol/lib/util.js:4:10)
    at NetworkDevice.wake (/usr/local/lib/node_modules/homebridge-wol/lib/network-device.js:176:15)
    at NetworkDevice.setOnline (/usr/local/lib/node_modules/homebridge-wol/lib/network-device.js:138:20)
    at On.emit (events.js:315:20)
    at On.EventEmitter.emit (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/EventEmitter.ts:42:22)
    at On.Characteristic._this.setValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:652:12)

### Notes
Please write about things that you did not see fit the above headers.

**Comment by AlexGustafsson** [comment]
The format for the MAC should be something like `12:34:56:78:90:AB`. What format do you have?

**Comment by mhuell** [comment]
The format looks like this: 12:3A:AB:4A:AA:A5

**Comment by AlexGustafsson** [comment]
That should work. Could you try lowercase: `12:3a:ab:4a:aa:a5`? Also, double and triple check your config so that there's no weird error in the MAC address, such as a simple space within the quotes or anything. I haven't actually seen this error before and we've been using the same WoL package since the start.

**Comment by AlexGustafsson** [comment]
The relevant code is here: https://github.com/agnat/node_wake_on_lan/blob/dd6404623ccf4bc55b0ba8788ba38312b55435a1/wake_on_lan.js#L16-L21.

```javascript
if (mac.length == 2 * mac_bytes + (mac_bytes - 1)) {
    mac = mac.replace(new RegExp(mac[2], 'g'), '');
  }
  if (mac.length != 2 * mac_bytes || mac.match(/[^a-fA-F0-9]/)) {
    throw new Error("malformed MAC address '" + mac + "'");
  }
```

Basically the length should either 12 or 18. Yours is 18. When we remove the third character (in your case `:`) we should have a length of 12. Yours is `123AAB4AAAA5`, which has the length 12. Lastly the MAC should match `/[^a-fA-F0-9]/` which yours does.

**Comment by mhuell** [comment]
I’ve checked the MAC address - tried lower case and upper case with and without “:” in between the characters before - tried different configurations. However I get every time the same error with my MAC address.

**Comment by AlexGustafsson** [comment]
This behaviour seems weird indeed. I have some questions for you, which might help narrow things down.

* What operating system are you on?
* What version of Node are you on?
* What version of homebridge-wol are you using?
* Can you check what version of `wake_on_lan` you're using? If you've installed homebridge-wol globally  `npm list -g | wake_on_lan` should work.

**Comment by mhuell** [comment]
os & os version: Raspbian GNU/Linux Buster (10)
    node version: v14.15.1
    homebridge version: v1.1.6
    homebridge-wol version: v4.3.0

Example: starting the windows client with etherwake works fine

**Comment by AlexGustafsson** [comment]
Everything seems up to date there. I can't find anything that's off. Unfortunately I cannot help you much further. This plugin uses the value given in the config and feeds it directly in to the [wake on lan](https://github.com/agnat/node_wake_on_lan#readme) package.

Check the following two things:
1. Your MAC address only contains the following characters: `0123456789ABCDEFabcdef`` (beyond any seperator such as `:`)
2. The MAC address that is being logged within the quotes (`Error: malformed MAC address 'MAC HERE'`) is exactly the value you have in your config

If you are able to verify both of those, I'm afraid you'll have to look for help elsewhere (such as the wake on lan package's repository). I really can't find anything else that's wrong.

There's been no change made to how the MAC address is handled, but for sanity you may want to try the newest version (#124).

**Comment by mhuell** [comment]
ok thank you so far. I will try it mysef maybe I'll get it!

**Comment by AlexGustafsson** [comment]
Please update this issue if you're able to solve it, it can help others facing the same issue in the future. Good luck!

**Comment by mhuell** [comment]
For now I build another solution where I don’t use that routine to check the mac address and pushing an wol command direkt over the Switch. That’s works fine for me. If I do further investigations to that issue I will let you know.

**Comment by AlexGustafsson** [comment]
Closing for now. Feel free to reopen if you come across this issue again.

**Comment by ndraiman** [comment]
First of all thank you for the great plugin.
I also encountered this issue, had the same problem as @mhuell, but I managed to solved it.

I copied the MAC address from my router web interface and made sure it fits the network adapter of the computer.
Once i confirmed that, the only other issue i could think of were invisible characters that were copied with the MAC address.

In short, typing the MAC address by hand solved the issue for me.
